include/engine/datafacade/datafacade_base.hpp

```diff
@@ -72,6 +72,10 @@ class BaseDataFacade
 
     virtual unsigned GetGeometryIndexForEdgeID(const unsigned id) const = 0;
 
+    virtual TurnPenalty GetWeightPenaltyForEdgeID(const unsigned id) const = 0;
```

Comment:
Is the type off here? Edge-Weights/Edge-Durations should be getting much larger than TurnPenalties. Limiting them here could be a possible overflow issue.